modules/tensor_mechanics/src/materials/ADMultiplePowerLawCreepStressUpdate.C

```diff
@@ -0,0 +1,141 @@
+    _activation_energy(getParam<std::vector<Real>>("activation_energy")),
+    _gas_constant(getParam<Real>("gas_constant")),
+    _start_time(getParam<Real>("start_time")),
+    _exponential(1),
+    _exp_time(0),
+    _stress_index(0),
+    _number_of_models(_m_exponent.size())
+{
+
+  if (!std::is_sorted(_stress_thresholds.begin(), _stress_thresholds.end()))
+    paramError("stress_thresholds", "Stress thresholds input must be ordered in increasing order");
+
+  if (_coefficient.size() != _n_exponent.size() || _coefficient.size() != _m_exponent.size() ||
+      _coefficient.size() != _activation_energy.size())
+    paramError(
+        "n_exponent",
+        "Inputs to ADMultiplePowerLawCreepStressUpdate creep models must have the same size");
+
+  if (_number_of_models != _stress_thresholds.size() - 1)
+    paramError("stress_thresholds",
+               "The number of creep models must be the number of stress thresholds minute. That "
+               "is, one creep model will be valid between two stress thresholds");
+
+  for (unsigned int i = 0; i < _number_of_models; i++)
+    if (_start_time < this->_app.getStartTime() && (std::trunc(_m_exponent[i]) != _m_exponent[i]))
+      paramError("start_time",
+                 "Start time must be equal to or greater than the Executioner start_time if a "
+                 "non-integer m_exponent is used");
+}
+
+void
+ADMultiplePowerLawCreepStressUpdate::computeStressInitialize(
+    const ADReal & effective_trial_stress, const ADRankFourTensor & /*elasticity_tensor*/)
+{
+  _stress_index = stressIndex(effective_trial_stress);
+
+  if (_temperature)
+    _exponential =
+        std::exp(-_activation_energy[_stress_index] / (_gas_constant * (*_temperature)[_qp]));
+
+  _exp_time = std::pow(_t - _start_time, _m_exponent[_stress_index]);
+}
+
+ADReal
+ADMultiplePowerLawCreepStressUpdate::computeResidual(const ADReal & effective_trial_stress,
+                                                     const ADReal & scalar)
+{
+  const ADReal stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  const ADReal creep_rate = _coefficient[_stress_index] *
+                            std::pow(stress_delta, _n_exponent[_stress_index]) * _exponential *
+                            _exp_time;
+  return creep_rate * _dt - scalar;
+}
+
+ADReal
+ADMultiplePowerLawCreepStressUpdate::computeDerivative(const ADReal & effective_trial_stress,
+                                                       const ADReal & scalar)
+{
+  const ADReal stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  const ADReal creep_rate_derivative =
+      -_coefficient[_stress_index] * _three_shear_modulus * _n_exponent[_stress_index] *
+      std::pow(stress_delta, _n_exponent[_stress_index] - 1.0) * _exponential * _exp_time;
+  return creep_rate_derivative * _dt - 1.0;
+}
+
+Real
+ADMultiplePowerLawCreepStressUpdate::computeStrainEnergyRateDensity(
```

Comment:
This is untested?

https://mooseframework.inl.gov/docs/PRs/18862/coverage_diff/modules.html#modules/tensor_mechanics/src/materials/ADMultiplePowerLawCreepStressUpdate.C

Reply:
Those lines should be covered now.

Reply:
I don't know if the coverage report is outdated, but it looks like `computeStrainEnergyRateDensity` is never called. Also the temperature dependence is not covered by tests. (all physics)

Reply:
The report is indeed outdated right now. It's the last thing to run.

Should all physics be covered?

Reply:
> Should all physics be covered?

I think it is worthwhile to aspire to cover all implemented physics with tests, yes. Do you disagree? 
However my "(all physics)" referred to all of your covered lines (in the outdated report) being actual physics, rather than assertions or off-normal conditions (which we should also cover by tests according to NQA standards).